src/scss/components/switch/_index.scss

```diff
@@ -0,0 +1,178 @@
+  justify-items: start;
+  grid-area: facade;
+  margin-top: 0.25rem; // FIXME: fix with proper inline styling
+  position: relative;
+  width: 4.25rem;
+  height: 2rem;
+
+  > input {
+    opacity: 0;
+    width: 0;
+    height: 0;
+  }
+}
+
+hx-switch-control > input[type="checkbox"] ~ label {
+  > hx-switch {
+    background-color: #ffffff;
+    border: 0.125rem solid #0c7c84;
+  }
+
+  > hx-switch[offlabel]::after {
+    content: attr(offlabel);
+    text-transform: uppercase;
+    margin-left: 1.688rem;
+    color: #0c7c84;
+    line-height: 1.75rem;
+    max-width: 1.813rem;
+    display: inline-block;
+    overflow: hidden;
+    white-space: nowrap;
+  }
+
+  > hx-switch::before {
+    background-color: #ffffff;
+    border: 0.125rem solid #0c7c84;
+    top: 0;
+    left: 0;
+  }
+}
+
+hx-switch-control > input[type="checkbox"]:checked ~ label {
+  > hx-switch {
+    background-color: #0c7c84;
+    @include hxSwitchControl($pseudo-state: focus);
+  }
+
+  > hx-switch[onlabel]::after {
+    content: attr(onlabel);
+    text-transform: uppercase;
+    margin-left: 0.625rem;
+    color: #ffffff;
+    max-width: 2.125rem;
+    display: inline-block;
+    overflow: hidden;
+    white-space: nowrap;
+  }
+
+  > hx-switch::before {
+    margin-left: 2.438rem;
+    height: 1.813rem;
+    top: 0;
+  }
+}
+
+.switch {
+  position: relative;
+  display: inline-block;
+  width: 4.25rem;
+  height: 2rem;
+}
+
+.slider {
+  position: absolute;
+  cursor: pointer;
+  top: 0;
+  left: 0;
+  right: 0;
+  bottom: 0;
+  -webkit-transition: 0.4s;
+  transition: 0.4s;
+}
+
+.slider:before {
+  position: absolute;
+  content: "";
+  height: 1.75rem;
+  width: 1.625rem;
+  left: 0.25rem;
+  bottom: 0.25rem;
+  -webkit-transition: 0.4s;
+  transition: 0.4s;
+}
+
+/* Rounded sliders */
+.slider.round {
+  border-radius: 2.125rem;
+}
+
+.slider.round:before {
+  border-radius: 50%;
+}
+
+// ========== DISABLED ==========
+hx-switch-control > input[type="checkbox"]:disabled {
+  &:checked {
+    ~ label > hx-switch {
+      @include hxSwitchControl(disabled);
+    }
+  }
+
+  ~ label > hx-switch {
+    @include hxSwitchControl(disabled, unchecked);
+  }
+}
+
+// --- start of error state css for hx-switch
+hx-switch-control {
+  .error {
+    &:checked {
+      ~ label > hx-switch {
+        @include hxSwitchControl(error, null, checked);
+        @include hxSwitchControl(error, focus);
+      }
+    }
+
+    ~ label > hx-switch {
+      @include hxSwitchControl(error, null, unchecked);
+      @include hxSwitchControl(error, focus);
+    }
+  }
+
+  // --- end of error state css for hx-switch
+
+  // --- start of hover css for hx-switch
+  #errorChkDemo:hover {
+    &:checked {
+      ~ label > hx-switch {
+        @include hxSwitchControl($pseudo-state: hover);
+      }
+    }
+
+    ~ label > hx-switch {
+      @include hxSwitchControl(
+        $pseudo-state: hover,
+        $checkBox-state: unchecked
+      );
+    }
+  }
+
+  #switchTest:not([disabled]):hover {
```

Comment:
Are these styles required for the demo or the Switch component?

Reply:
Yes, these styles are required for switch component . but, as discussed i will change class to attribute in input element